pkg/controllers/provisioning/provisioner.go

```diff
@@ -217,6 +217,10 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNod
 	instanceTypes := map[string][]*cloudprovider.InstanceType{}
 	domains := map[string]sets.Set[string]{}
 	for _, nodePool := range nodePoolList.Items {
+		nodePoolReady := nodePool.StatusConditions().Get(v1beta1.ConditionTypeNodePoolReady)
+		if !nodePoolReady.IsTrue() {
+			return nil, fmt.Errorf("nodepool not ready, %s", nodePoolReady.Message)
```

Comment:
Do we want to produce an error here if a NodePool is not ready? I think we should log but skip the NodePool for provisioning